command/agent/config/config.go

```diff
@@ -193,8 +188,19 @@ func LoadConfig(path string) (*Config, error) {
 		return nil, errwrap.Wrapf("error parsing 'vault':{{err}}", err)
 	}
 
-	if err := parseRetry(result, list); err != nil {
-		return nil, errwrap.Wrapf("error parsing 'retry': {{err}}", err)
+	if result.Vault == nil {
```

Comment:
Can this ever be nil if we're always assigning it within `parseVault` (L236, `result.Vault = &v`)?

Reply:
It can be nil if there's no `vault{}` stanza.

Reply:
Ah, I see. We could perform the assignment first within that func (similar to what we do [in here](https://github.com/hashicorp/vault/blob/master/internalshared/configutil/telemetry.go#L167)), but I'll leave that up to you.